node/service/src/lib.rs

```diff
@@ -211,3 +205,83 @@ impl<C: Components> ::std::ops::Deref for Service<C> {
 		&self.inner
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use std::time;
+	use std::sync::Arc;
+	use parking_lot::RwLock;
+	use {service, service_test, Factory, chain_spec};
+	use consensus::{self, OfflineTracker};
+	use primitives::ed25519;
+	use node_primitives::{BlockId, Header, Hash, Block};
+	use runtime_primitives::traits::{Hash as HashT, BlakeTwo256};
+	use node_api::Api;
+	use bft::{self, Proposer};
+	use rhododendron;
+
+	#[test]
+	fn test_connectivity() {
+		service_test::connectivity::<Factory>(chain_spec::integration_test_config());
+	}
+
+	#[test]
+	fn test_sync() {
+		let alice = Arc::new(ed25519::Pair::from_seed(b"Alice                           "));
+		let bob = Arc::new(ed25519::Pair::from_seed(b"Bob                             "));
+		let authorities = vec![alice.clone(), bob.clone()];
+		let validators = vec![alice.public().0.into(), bob.public().0.into()];
+		let offline = Arc::new(RwLock::new(OfflineTracker::new()));
+		let block_factory = |service: &<Factory as service::ServiceFactory>::FullService| {
+			let parent_hash = service.client().info().unwrap().chain.best_hash;
+			let parent_number = service.client().info().unwrap().chain.best_number;
+			let parent_id = BlockId::hash(parent_hash);
+			let random_seed = service.client().random_seed(&parent_id).unwrap();
+			let random_seed = BlakeTwo256::hash(&*random_seed);
+			let timestamp = time::SystemTime::now().duration_since(time::UNIX_EPOCH).unwrap().as_secs();
+			let proposer = consensus::Proposer {
+				client: service.client().clone(),
+				start: time::Instant::now(),
+				local_key: alice.clone(),
+				parent_hash,
+				parent_id: parent_id,
+				parent_number: parent_number,
+				random_seed,
+				transaction_pool: service.transaction_pool().clone(),
+				offline: offline.clone(),
+				validators: validators.clone(),
+				minimum_timestamp: timestamp,
+			};
+			let block = proposer.propose().expect("Error making test block");
+			let justification = justify(&block.header, authorities.as_slice());
```

Comment:
If this waits until after the import queue stuff, we won't have to add additional reliance on rhododendron

Reply:
Yeah, I expect this bit to be reworked in the near future. Still I would prefer for this PR not to be blocked on future work.